src/coreclr/pal/src/sharedmemory/sharedmemory.cpp

```diff
@@ -221,16 +289,32 @@ int SharedMemoryHelpers::Open(LPCSTR path, int flags, mode_t mode)
             return -1;
 
         case ENAMETOOLONG:
-            throw SharedMemoryException(static_cast<DWORD>(SharedMemoryError::NameTooLong));
+            sharedMemoryError = SharedMemoryError::NameTooLong;
+            break;
 
         case EMFILE:
         case ENFILE:
         case ENOMEM:
-            throw SharedMemoryException(static_cast<DWORD>(SharedMemoryError::OutOfMemory));
+            sharedMemoryError = SharedMemoryError::OutOfMemory;
+            break;
 
         default:
-            throw SharedMemoryException(static_cast<DWORD>(SharedMemoryError::IO));
+            sharedMemoryError = SharedMemoryError::IO;
+            break;
+    }
+
+    if (sharedMemoryError != SharedMemoryError::NameTooLong)
```

Comment:
Is this `if` necessary? It seems we should not be getting that and if we do, we would like to know why.

Reply:
The `NameTooLong` case already returns `ERROR_FILENAME_EXCED_RANGE`, for which a `PathTooLongException` exception is thrown, which seemed sufficient to me, but I guess it would be nice to see what the path was. I'll update.

Reply:
It looks like the exception thrown already includes the mutex name, that should be sufficient for this case.